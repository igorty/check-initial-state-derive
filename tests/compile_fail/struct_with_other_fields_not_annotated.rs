use check_initial_state_derive::CheckInitialState;

/// Checks case when struct contains non&#x2011;`Option` fields which are not
/// annotated with `ignore_field`.
fn main() {
	Struct { option: Some(10), integer: 10 }.check_initial_state();
}

#[derive(CheckInitialState)]
struct Struct {
	#[ignore_field]
	option: Option<i32>,
	integer: i32,
}
