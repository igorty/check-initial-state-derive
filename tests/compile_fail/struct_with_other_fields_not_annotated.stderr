error[E0308]: mismatched types
 --> $DIR/struct_with_other_fields_not_annotated.rs:9:10
  |
9 | #[derive(CheckInitialState)]
  |          ^^^^^^^^^^^^^^^^^ expected enum `std::option::Option`, found `i32`
  |
  = note: expected reference `&std::option::Option<_>`
             found reference `&i32`
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
